src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ModuleInitializerAttributeShouldNotBeUsedInLibrariesTests.cs

```diff
@@ -0,0 +1,207 @@
+                        }}
+                        "
+                    }
+                },
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                MarkupOptions = MarkupOptions.UseFirstDescriptor
+            }.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("public", "public", false)]
+        [InlineData("public", "internal", false)]
+        [InlineData("internal", "public", false)]
+        [InlineData("internal", "internal", false)]
+        [InlineData("public", "public", true)]
+        [InlineData("public", "internal", true)]
+        [InlineData("internal", "public", true)]
+        [InlineData("internal", "internal", true)]
+        public async Task CA2255ModuleInitializerOnMethod_Suppressed(string classModifier, string methodModifier, bool useAsync = false)
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                    Sources = {
+                        $@"
+                        #pragma warning disable CA2255
+                        {classModifier} class CA2255ModuleInitializerOnMethod
+                        {{
+                            [System.Runtime.CompilerServices.ModuleInitializer]
+                            {methodModifier} static {(useAsync ? "async" : "")} void Initialize() {{ }}
+                        }}
+                        #pragma warning restore CA2255
+                        "
+                    }
+                },
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                MarkupOptions = MarkupOptions.UseFirstDescriptor
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task CA2255DoesNotApply_ToPrivateMethods()
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                    Sources = {
+                        @$"
+                        public class CA2255ModuleInitializerOnMethod
+                        {{
+                            [{{|CS8814:System.Runtime.CompilerServices.ModuleInitializer|}}]
+                            private static void Initialize() {{ }}
+                        }}
+                        "
+                    }
+                },
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                MarkupOptions = MarkupOptions.UseFirstDescriptor
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task CA2255DoesNotApply_ToNonVoidMethods()
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                    Sources = {
+                        @$"
```

Comment:
interpolation not needed. This (and other test feedback) applies to other places. I won't add extra repeated comments to keep things cleaner and easier to track feedback.

Reply:
Thanks. I had consciously kept the interpolation in place because it kept the code copy-pastable across the tests. But I'll update it.